packages/compartment-mapper/src/import-hook.js

```diff
@@ -718,14 +717,38 @@ export function makeImportNowHookMaker(
             });
             return exitRecord;
           }
+          throw Error(
+            `Cannot find external module ${q(
+              moduleSpecifier,
+            )} in package at ${packageLocation}`,
+          );
+        } else {
+          throw Error(
+            `Cannot find external module ${q(
+              moduleSpecifier,
+            )} from package at ${packageLocation}; try providing an importNowHook`,
+          );
         }
-        throw Error(
-          `Cannot find external module ${q(
-            moduleSpecifier,
-          )} in package ${packageLocation}`,
-        );
+      };
+
+      // many dynamically-required specifiers will be absolute paths owing to use of `require.resolve()` and `path.resolve()`
+      if (isAbsolute(moduleSpecifier)) {
+        const record = findRedirect({
+          compartmentDescriptor,
+          compartmentDescriptors,
+          compartments,
+          absoluteModuleSpecifier: moduleSpecifier,
+        });
+        if (record) {
+          return record;
+        }
+      } else if (moduleSpecifier !== '.' && !moduleSpecifier.startsWith('./')) {
+        // could be a builtin, which means we should not bother bouncing on the trampoline to find it.
+        return tryExitModuleImportNowHook();
       }
 
+      // we might have an absolute path here, but it might be within the compartment, so
+      // we will try to find it.
       const candidates = nominateCandidates(moduleSpecifier, searchSuffixes);
 
       const record = syncTrampoline(
```

Comment:
AFAIR going into `chooseModuleDescriptor` with an absolute path can result in an error that got us to find this issue in the first place. Some of the nested functionality is not ready to handle absolute paths. If the path is within the current compartment, we should be able to convert it to relative.

Reply:
so your recommendation would be to convert it to relative if the path is in the current compartment?

Reply:
if and only if, yes.

Reply:
this is done.